sdc/datatypes/hpat_pandas_functions.py

```diff
@@ -38,60 +38,261 @@
+        # Column and Index Locations and Names
+        header="infer",
+        names=None,
+        index_col=None,
+        usecols=None,
+        squeeze=False,
+        prefix=None,
+        mangle_dupe_cols=True,
+        # General Parsing Configuration
+        dtype=None,
+        engine=None,
+        converters=None,
+        true_values=None,
+        false_values=None,
+        skipinitialspace=False,
+        skiprows=None,
+        skipfooter=0,
+        nrows=None,
+        # NA and Missing Data Handling
+        na_values=None,
+        keep_default_na=True,
+        na_filter=True,
+        verbose=False,
+        skip_blank_lines=True,
+        # Datetime Handling
+        parse_dates=False,
+        infer_datetime_format=False,
+        keep_date_col=False,
+        date_parser=None,
+        dayfirst=False,
+        cache_dates=True,
+        # Iteration
+        iterator=False,
+        chunksize=None,
+        # Quoting, Compression, and File Format
+        compression="infer",
+        thousands=None,
+        decimal=b".",
+        lineterminator=None,
+        quotechar='"',
+        # quoting=csv.QUOTE_MINIMAL,  # not supported
+        doublequote=True,
+        escapechar=None,
+        comment=None,
+        encoding=None,
+        dialect=None,
+        # Error Handling
+        error_bad_lines=True,
+        warn_bad_lines=True,
+        # Internal
+        delim_whitespace=False,
+        # low_memory=_c_parser_defaults["low_memory"],  # not supported
+        memory_map=False,
+        float_precision=None,
+    """
+
+    # read_csv can infer result DataFrame type from file or from params
+
+    # for inferring from file this parameters should be literal or omitted
+    infer_from_file = all([
+        isinstance(filepath_or_buffer, types.Literal),
+        isinstance(sep, (types.Literal, types.Omitted)) or sep == ',',
+        isinstance(delimiter, (types.Literal, types.Omitted)) or delimiter is None,
+        isinstance(names, (types.Tuple, types.Omitted, type(None))),
+        isinstance(usecols, (types.Tuple, types.Omitted, type(None))),
+        isinstance(skiprows, (types.Literal, types.Omitted)) or skiprows is None,
+    ])
+
+    # for inference from params dtype and (names or usecols) shoud present
+    # names, dtype and usecols should be literal tuples after rewrite pass (see. RewriteReadCsv)
+    # header not supported
+    infer_from_params = all([
+        isinstance(dtype, types.Tuple),
+        any([
+            isinstance(names, types.Tuple) and isinstance(usecols, types.Tuple),
+            isinstance(names, types.Tuple) and isinstance(usecols, (types.Omitted, type(None))),
+            isinstance(names, (types.Omitted, type(None))) and isinstance(usecols, types.Tuple),
+        ]),
+        isinstance(header, types.Omitted) or header == 'infer',
+    ])
+
+    # cannot create function if parameters provide not enough info
+    assert infer_from_file or infer_from_params
```

Comment:
Could you please replace assert with exception?

Reply:
Fixed